Flipper/Packages/UI/Sources/Common/Image/SVG/SVGParser.swift

```diff
@@ -0,0 +1,200 @@
+import Foundation
+
+class SVGParser: XMLParser, XMLParserDelegate {
+    private(set) var commands: [SVGCommand] = []
+    private(set) var height: Double?
+    private(set) var width: Double?
+
+    private var isParsingGroup: Bool = true
+
+    override init(data: Data) {
+        super.init(data: data)
+        self.delegate = self
+    }
+
+    func parser(
+        _ parser: XMLParser,
+        didStartElement elementName: String,
+        namespaceURI: String?,
+        qualifiedName qName: String?,
+        attributes attributeDict: [String: String] = [:]
+    ) {
+        switch elementName {
+        case "path":
+            guard isParsingGroup else { return }
+            parsePathElement(attributes: attributeDict)
+        case "rect":
+            guard isParsingGroup else { return }
+            parseRectElement(attributes: attributeDict)
+        case "defs": // Stop parser out of main svg content
+            isParsingGroup = false
+        case "svg":
+            parseSvgMetaElement(attributes: attributeDict)
+        default: break
+        }
+    }
+
+    private func parseSvgMetaElement(attributes: [String: String]) {
+        guard
+            let width = attributes["width"], let width = Double(width),
+            let height = attributes["height"], let height = Double(height)
+        else { return }
+
+        self.height = height
+        self.width = width
+    }
+
+    private func parsePathElement(attributes: [String: String]) {
+        guard let dAttribute = attributes["d"] else { return }
+
+        let rawCommands = parseRawCommands(data: dAttribute)
+        let svgCommands = rawCommands.map(parseSVGCommand)
+
+        self.commands += svgCommands
+    }
+
+    private func parseRawCommands(data: String) -> [String] {
+        let supportedTags = "MmLlHhVvZzC"
+
+        var result = [String]()
+        var currentCommand = ""
+
+        for character in data {
+            if supportedTags.contains(character) {
+                if !currentCommand.isEmpty {
+                    result.append(currentCommand)
+                }
+                currentCommand = String(character)
+            } else {
+                currentCommand.append(character)
+            }
+        }
+
+        if !currentCommand.isEmpty {
+            result.append(currentCommand)
+        }
+
+        return result
+    }
+
+    private func parseSVGCommand(command: String) -> SVGCommand {
+        guard let commandCharacter = command.first else { return .unknown }
+
+        let value = command
+            .dropFirst()
+            .split(separator: " ")
+            .compactMap{ Double($0) }
+
+        switch commandCharacter {
+        case "M":
```

Comment:
please check if the **value** has enough elements to prevent possible **index out of range** error